src/main/kotlin/uk/gov/justice/digital/hmpps/nomisprisonerapi/config/NomisPrisonerApiExceptionHandler.kt

```diff
@@ -50,6 +51,20 @@ class NomisPrisonerApiExceptionHandler {
       )
   }
 
+  @ExceptionHandler(DataNotFoundException::class)
+  fun handleBadRequestException(e: Exception): ResponseEntity<ErrorResponse?>? {
+    log.info("Bad request: {}", e.message)
+    return ResponseEntity
+      .status(BAD_REQUEST)
+      .body(
+        ErrorResponse(
+          status = BAD_REQUEST,
+          userMessage = "Bad request: ${e.message}",
+          developerMessage = e.message
+        )
+      )
+  }
+
```

Comment:
Is 400 rather than 404 a general pattern we are going to follow in this scenario?

Reply:
Well this is what you get when the request contains a nonexistant person id, prison or room.

Reply:
Which feels better than a 404 since an incorrect roomId for instance is really a bad request. Perhaps the exception is perhaps when the actually offender is not found